submissions/Valerii49/side_menu/js/script.js

```diff
@@ -0,0 +1,151 @@
+"use strict";
+import sitesArr from '../data/attractions.js';
+
+document.addEventListener("DOMContentLoaded", InitPage(sitesArr));
+
+function InitPage(data) {
+	const menuList = document.querySelector('.menu__list');
+	const h1Link = document.querySelector('.h1__link');
+	//**********Init Sidebar**********
+	const menuLinks = data
+		.slice(1)
+		.map(({ id, sightName }) => `<li><a href="#" class="menu__link" data-id="${id}">${sightName}</a></li>`)
+		.join('');
+	menuList.innerHTML = menuLinks;
+	//**********for hover effect**********
+	const menuItems = document.querySelectorAll('.menu__link');
+	// running letters
+	const stringToSpans = (item, delay) => {
+		item.innerHTML = item.innerText
+			.split('')
+			.map((letter, i) => `<span style="transition-delay:${i * delay}ms;">${letter}</span>`)
+			.join('');
+	}
+	stringToSpans(h1Link, 10);
+	menuItems.forEach(element => {
+		stringToSpans(element, 30);
+	});
+	// 
+	const mediaQuery = window.matchMedia('(min-width: 48rem)');
+	mediaQuery.addEventListener('change', (e) => {
+		stringToSpans(h1Link, 10);
+		mediaQuery.matches && menuItems.forEach(element => { stringToSpans(element, 30) });
+	});
+	const page = document.querySelector('.page');
+
+	page.addEventListener("click", handlerClick);
+}
+
+function handlerClick(e) {
+	const targetElement = e.target;
+	const currentItem = targetElement.closest('.menu__link') ?? targetElement.closest('.h1__link');
+
+	const currentImage = targetElement.closest('.main-module__image-gallery');
+
+	if (currentItem) {
+		if (document.documentElement.closest(".menu-open")) {
+			menuBurgerToggle();
+		}
+		const activeItem = document.querySelector('._active');
+		if (currentItem !== activeItem) {
+			activeItem.classList.remove('_active');
+			currentItem.classList.add('_active');
+			const itemId = currentItem.dataset.id - 1;
+			buildPage(sitesArr[itemId]);
+		}
+	}
+	else if (currentImage) {
```

Comment:
If you already implement change from my comment above, we can use same approach here with this else if, we can remove it and also use if (currentImage && currentImage !== whatever)